Latest commits broke the internal api.json generator

<!-- Please search existing issues for potential duplicates before filing yours:
https://github.com/godotengine/godot/issues?q=is%3Aissue
-->

**Godot version:**
<!-- Specify commit hash if non-official. -->
broken c2c11fc063e723c41579e6307e40fd16ac5334c4
works v3.1 release

**OS/device including version:**
Linux

**Steps to reproduce:**
godot32 --gdnative-generate-json-api api.json

**A few excerpts:**

https://docs.godotengine.org/en/latest/classes/class_undoredo.html#class-undoredo-method-add-do-method

```
@@ -8401,7 +8464,7 @@
                                "arguments": [
                                        {
                                                "name": "object",
-                                               "type": "Object",
+                                               "type": "",
                                                "has_default_value": false,
                                                "default_value": ""
                                        },
@@ -8426,7 +8489,7 @@
                                "arguments": [
                                        {
                                                "name": "object",
-                                               "type": "Object",
+                                               "type": "",
                                                "has_default_value": false,
                                                "default_value": ""
                                        },


clang++ -o src/gen/RigidBody.o -c -fPIC -g -std=c++14 -Wwrite-strings -Og -m64 -I. -Igodot_headers -Iinclude -Iinclude/gen -Iinclude/core src/gen/RigidBody.cpp
In file included from src/gen/RigidBody.cpp:1:
In file included from include/gen/RigidBody.hpp:12:
In file included from include/gen/PhysicsBody.hpp:11:
include/gen/CollisionObject.hpp:14:1: error: declaration of anonymous class must be a definition
class ;
^
include/gen/CollisionObject.hpp:14:1: warning: declaration does not declare anything [-Wmissing-declarations]
include/gen/CollisionObject.hpp:30:28: error: C++ requires a type specifier for all declarations
        void _input_event(const  *camera, const Ref<InputEvent> event, const Vector3 click_position, const Vector3 click_normal, const int64_t shape_idx);
                          ~~~~~   ^
```

--- a/modules/gdnative/nativescript/api_generator.cpp
+++ b/modules/gdnative/nativescript/api_generator.cpp
@@ -323,6 +323,9 @@ List<ClassAPI> generate_c_api_classes() {
 						arg_type = "Variant";
 					} else if (arg_info.type == Variant::OBJECT) {
 						arg_type = arg_info.class_name;
+						if (arg_type == "") {
+							arg_type = Variant::get_type_name(arg_info.type);
+						}
 					} else {
 						arg_type = Variant::get_type_name(arg_info.type);
 					}

